### ISSUE TITLE
Not Finding All WeMo Devices In Latest Version 0.6.9

### ISSUE THREAD

**Issue by mmaxwell5** [rudders/homebridge-platform-wemo#35]
I recently updated to the latest version of homebridge-platform-wemo (0.6.9, was on 0.6.6).  Since updating, homebridge will not find all my wemo devices.  I have to restart the service 20 + times to find them all.  

I do not have any issues using the WeMo app with connectivity to the light switches.  I usually test that after it fails to find the device in homebridge and it works right away in the WeMo app.  It also seems to be a different device not being discovered on every start of homebridge.  I currently have 6 devices configured.  

Thank you in advance for any help anyone is able to provide.  Love homebridge and WeMo support.   

homebridge.log:

```
Loaded plugin: homebridge-nest
Registering platform 'homebridge-nest.Nest'
---
Loaded plugin: homebridge-philipshue
Registering platform 'homebridge-philipshue.PhilipsHue'
---
Loaded plugin: homebridge-platform-wemo
Registering platform 'homebridge-platform-wemo.BelkinWeMo'
---
Loaded config.json with 0 accessories and 3 platforms.
---
Loading 3 platforms...
Initializing Nest platform...
Fetching Nest devices.
[Philips Hue] Initializing PhilipsHue platform...
[Philips Hue] PhilipsHue Platform Plugin Version 0.0.2
[Philips Hue] Fetching Philips Hue lights...
[Philips Hue] Attempting to discover Philips Hue bridge with meethue.com...
[WeMo Platform] Initializing BelkinWeMo platform...
[WeMo Platform] Wemo Platform Plugin Loaded
[WeMo Platform] Fetching the Wemo Accessories, expecting 6 and will wait 120 seconds to find them.
Loading 0 accessories...
[WeMo Platform] Found: Office Light, type: lightswitch
[WeMo Platform] Office Light is true
[WeMo Platform] Found: Office Light, type: lightswitch
[WeMo Platform] Office Light is true
[WeMo Platform] Discovered 1 accessories of 6
[WeMo Platform] Found: Bonus Room Light, type: lightswitch
[WeMo Platform] Bonus Room Light is false
[WeMo Platform] Discovered 2 accessories of 6
[WeMo Platform] Found: Bedroom Light, type: lightswitch
[WeMo Platform] Bedroom Light is false
[WeMo Platform] Discovered 3 accessories of 6
[WeMo Platform] Found: CoffeeMaker, type: CoffeeMaker
[WeMo Platform] CoffeeMaker is false
[WeMo Platform] Discovered 4 accessories of 6
[WeMo Platform] Found: Bedroom Fan, type: lightswitch
[WeMo Platform] Bedroom Fan is true
[WeMo Platform] Discovered 5 accessories of 6
[WeMo Platform] Office Light binaryState: on
[WeMo Platform] Bonus Room Light binaryState: off
[WeMo Platform] Bedroom Light binaryState: off
[WeMo Platform] Bedroom Fan binaryState: on
[Philips Hue] Philips Hue bridges found:
        192.168.50.66 - 001788fffe242ee9
[Philips Hue] Save the Philips Hue bridge ip address 192.168.50.66 to your config to skip discovery.
[Philips Hue] Initializing platform accessory 'Front Porch Right'...
[Philips Hue] Initializing platform accessory 'Living Room Lamp'...
[Philips Hue] Initializing platform accessory 'Front Porch Left'...
[Philips Hue] Initializing platform accessory 'Garage Right'...
[Philips Hue] Initializing platform accessory 'Extra'...
[Philips Hue] Initializing platform accessory 'Garage Left'...
[Philips Hue] Initializing platform accessory 'Stairway'...
[Philips Hue] Initializing platform accessory 'TV Light'...
Temperature unit for Downstairs is: Fahrenheit
Current temperature for Downstairs is: 71 F
Current humidity for Downstairs is: 40%
Target temperature for Downstairs is: 66 F
Target heating for Downstairs is: Heating
Away for Downstairs is: true
Temperature unit for Upstairs is: Fahrenheit
Current temperature for Upstairs is: 70 F
Current humidity for Upstairs is: 40%
Target temperature for Upstairs is: 74 F
Target heating for Upstairs is: Cooling
Away for Upstairs is: true
Initializing platform accessory 'Downstairs'...
Initializing platform accessory 'Upstairs'...
[WeMo Platform] We have timed out and only discovered 5 of the specified 6 devices - try restarting homebridge or increasing timeout in$
[WeMo Platform] and you have indicated you'd like to keep your HomeKit config safe so we're crashing out
```

**Comment by rudders** [comment]
How complex is your lan? Seems like SSDP is having a hard time finding your devices. Does your homebridge server have multi paths to the network by chance (wifi and internet)?

**Comment by mmaxwell5** [comment]
I have multiple VLANs setup but my homebridge server and all my WeMo
devices are on the same LAN.  Looked in the switch and did not find
anything that would be blocking  SSDP.

For testing I move everything to a "dumb" apple router to keep it simple.
Same result.  I am having to restart the service 20 + times to get all the
devices discovered.

**Comment by blackjack75** [comment]
Are your devices on? My Wemo Insight Will only be discovered if the switch is on. Otherwise it is invisible.

I assume there is some technical reason for this (I haven't had a look at the code yet) but I wonder if some caching could be done to retrieve faster devices that were previously discovered and hopefully rediscover devices even if they are off.

I am in a situation currently where if I reboot my Raspberry pi running HomeBridgeduring the day I lose control of the light as it becomes invisible when it's off.

**Comment by rudders** [comment]
The re is a new version coming based on plugin-2 model recently promoted that will cache devices and try and reconnect. Still need to discover the devices in the first place though. This is handled by node-ssdp so am unsure why your discovery is so intermittent! My homebridge can take 10 restarts at times and One at others. I use pm to automate that though so it doesn't worry me too much. WEMO forums are full of issues even with the WEMO app.

**Comment by blackjack75** [comment]
Excellent, looking forward to such update! With caching hopefully I will need to turn no the light only on first detection.

Thanks for the great work!

**Comment by devbobo** [comment]
This issue should be resolved with the new `1.0.0` release
